### ISSUE TITLE
IDs not available from old Blockstack PK with Stacks Wallet

### ISSUE THREAD

**Issue by oluapocram** [leather-io/extension#1097]
When trying to restore a Blockstack PK into the Stacks Browser Plugin, no IDs are being shown. I've confirmed multiple times the PK is correct and restoring it on an older blockstack browser shows the IDs. 

```
Stacks Wallet 2.2.0
Brave | 1.23.44 Chromium: 89.0.4389.90 (Official Build) dev (x86_64)
OS | macOS Version 11.2.3 (Build 20D91)
JavaScript | V8 8.9.255.20
```

Steps to reproduce: 

1. Restore BS PK on https://browser.blockstack.org/ (there are ~15 IDs with this PK, but only showing 3 at the moment)

2. ie. marco.id is owned by https://explorer.legacy.blockstack.org/address/1LQxmuBK8hk6AFd6wENUKvKTSQRiTb2Z9j

3. Restoring the same BS PK with Stacks Wallet: 

4. Stacks explorer shows nothing associated with this SP: https://explorer.stacks.co/address/SPN7P13R7VPZG7QRH5KTNQF06W5FVTPH7QBDG0PN

**Comment by markmhendrickson** [comment]
I'm seeing similarly odd behavior with my own key, in that the same list of usernames shows up between the Stacks Wallet extension and Blockstack Connect on [app.blockstack.org](https://app.blockstack.org/) – but a different one shows up in the Blockstack Browser (with some of the same usernames included, but others missing). See screenshots below.

@hstove I believe we reviewed my case here a while back and spotted issues related to previously registered IDs on non-`.id.blockstack` subdomains, but it doesn't appear that we captured them anywhere. Do you recall what they were and if they apply to @oluapocram's case here as well?

@oluapocram what happens in the extension if you select "Create an account" a few more times? Do any of your `.id` usernames show up by chance further down the list?

I've created a related issue here for displaying usernames on the new Explorer: https://github.com/blockstack/explorer/issues/415

## Stacks Wallet

<img width="419" alt="Screen Shot 2021-03-24 at 13 46 38" src="https://user-images.githubusercontent.com/28991/112312814-81339300-8ca7-11eb-9e71-de5acd27153d.png">

<img width="418" alt="Screen Shot 2021-03-24 at 13 46 45" src="https://user-images.githubusercontent.com/28991/112312822-8395ed00-8ca7-11eb-9b87-ab8ff3c1f2db.png">

## Blockstack Connect

<img width="498" alt="Screen Shot 2021-03-24 at 13 46 51" src="https://user-images.githubusercontent.com/28991/112312838-8690dd80-8ca7-11eb-9927-2301e27239ea.png">

## Blockstack Browser

<img width="742" alt="Screen Shot 2021-03-24 at 13 46 27" src="https://user-images.githubusercontent.com/28991/112312812-809afc80-8ca7-11eb-984d-48996b67d894.png">
<img width="796" alt="Screen Shot 2021-03-24 at 13 46 31" src="https://user-images.githubusercontent.com/28991/112312803-7ed13900-8ca7-11eb-9a98-97630e19b0e6.png">

**Comment by oluapocram** [comment]
@markmhx I've added 7 more accounts - I don't see any new IDs popping up.

**Comment by hstove** [comment]
Hey @oluapocram, your IDs only show up automatically in the web wallet if you've used the authenticator at `app.blockstack.org` in Stacks 1.0 days. This is because the web wallet is not yet setup to query for BNS names on Stacks 2.0 - those api's are being deployed to mainnet nodes soon. The IDs only show up for other secret keys because `app.blockstack.org` introduced an encrypted "wallet config" - this is a file that simply caches the IDs you've used in the past. This allows us to "restore" all your IDs very quickly when you restore your wallet.

TLDR: we should be able to restore your IDs once BNS APIs are deployed to mainnet nodes. They were recently deployed to testnet, so it hopefully should not be too far off.

**Comment by markmhendrickson** [comment]
@hstove Aren't `app.blockstack.org` and the web wallet alike querying Stacks 1.0 username data using `core.blockstack.org` currently? And does it matter whether @oluapocram registered these usernames with `app.blockstack.org` or the Blockstack Browser previously?

**Comment by hstove** [comment]
It doesn't matter where the name was registered - only if there is a "wallet config" that exists. But, no, `app.blockstack.org` were configured to stop querying names from `core.blockstack.org` at the 2.0 launch. I think at the time there was less clarity around how long `core.blockstack.org` would be maintained, so all dependencies on it were removed from our apps.

**Comment by oluapocram** [comment]
@hstove @markmhx thanks for the update/clarification. 

To clarify: 

- Is there a way to know which Account SPxxx ID maps to what ID manually? 
- Once the BNS APIs are available on mainnet, are there changes needed in the Stacks Web Wallet to query the BNS APIs, or will it just work as expected?

**Comment by markmhendrickson** [comment]
@oluapocram My apologies for missing your last comment above from April. Is this issue still outstanding for you?

**Comment by oluapocram** [comment]
hey @markmhx - I just tried using the Hiro Wallet (2.14.0), restoring with my original BS seeds, I still don't see the IDs. 

Is this suppose to be fixed, or is there something you'd like me to test?

**Comment by markmhendrickson** [comment]
I suspect that we may have fixed the original caching problem here, only to run into this problem regarding a change in derivation paths: https://github.com/blockstack/stacks-wallet-web/issues/1412

@aulneau do you suspect the same here given the above details?

**Comment by aulneau** [comment]
Yes, it's very likely

**Comment by markmhendrickson** [comment]
@oluapocram Thanks for your patience here. Are you still missing your IDs in the latest version (v3.2.0)? 

We're taking a renewed look at legacy username issues in the hopes that we can resolve them for good.

**Comment by oluapocram** [comment]
@markmhx with 3.2.0... still seeing just wallet #'s. I can provide more details via DM on Discord if needed. 

<img width="161" alt="image" src="https://user-images.githubusercontent.com/28657119/155815095-4c482eb7-b8ea-4a3c-9ba9-9a7106717005.png">

**Comment by markmhendrickson** [comment]
@oluapocram Thanks for the response. On second thought here, the display of your IDs these days is probably affected by the following bug for which we have a fix preparing for release: https://github.com/hirosystems/stacks-wallet-web/issues/2241

I'll follow up once it's released so you can test again then. 🙏

**Comment by andresgalante** [comment]
Since #2241 has been fixed, I'll close this issue

@oluapocram please let me know if we need to reopen it.

**Comment by markmhendrickson** [comment]
I'd like to keep this one open since I believe it was an issue from before the introduction of #2241. Once the fix is live, we can ask @oluapocram to test again.

**Comment by markmhendrickson** [comment]
@oluapocram Mind testing again with the latest version available (v3.6.0)?

**Comment by markmhendrickson** [comment]
I'll close and we can reopen if this is reported to continue as a problem with the latest production version.
